clients/venice-client/src/main/java/com/linkedin/venice/fastclient/meta/AbstractStoreMetadata.java

```diff
@@ -36,6 +36,16 @@ public int getPartitionId(int version, byte[] key) {
     return getPartitionId(version, ByteBuffer.wrap(key));
   }
 
+  /**
+   * TODO: A quick question: Will it be a better way to have a list of all replicas in the caller of this function by calling getReplicas()
+   *  and remove the ones that are already used in the caller and this function is used as a wrapper to just call
+   *  routingStrategy.getReplicas() and get requiredReplicaCount number of replicas from that list? or this is intended as we are
+   *  optimizing for the first try and not for retries?
+   *
+   *  If the intention is not to optimize for the first try and not for retries, have a couple of questions on the implementation:
+   *  1. Calling getReplicas() for every retry => Is this necessary to get the updated metadata and not have the old stale metadata?
+   *  2. loop and check whether some replica is already tried => is this okay as the number of replicas is very small?
+   */
   @Override
```

Comment:
The intention is to hit the implementation details of `getReplicas`, so that the implementor can decide whether the original route will be excluded or not.
For example, the underlying implementation may return more replicas than asked for other purposes, and you can check the `ClientRoutingStrategy` impl.

Reply:
` QQ based on above TODO: Looks like the retry is being scheduled once via scheduler (LONG_TAIL_RETRY) and
If the intention is not to optimize for the first try and not for retries, have a couple of questions on the implementation:
   *  1. Calling getReplicas() for every retry => Is this necessary to get the updated metadata and not have the old stale metadata?
   *  2. loop and check whether some replica is already tried => is this okay as the number of replicas is very small?
`
For 1 , I guess its not necessary but why wouldn't you. We do want to get the latest metadata since its possible that new replicas got added between the requests. 
For 2 Its just filtering to remove the excluded instances. it won't be a performance issue since the number of replicas is under is small

Reply:
Makes sense. Thanks.